metainsuranceorg.py

```diff
@@ -433,3 +441,34 @@ def process_newrisks_insurer(self, risks_per_categ, number_risks_categ, acceptab
 
         return not_accepted_risks
 
+
+    def market_permanency(self, time):     #This method determines whether an insurer or reinsurer stays in the market. If it has very few risks underwritten or too much cash left for TOO LONG it eventually leaves the market.
+                                                      # If it has very few risks underwritten it cannot balance the portfolio so it makes sense to leave the market.
+        cash_left_by_categ = np.asarray(self.cash_left_by_categ)
+
+        avg_cash_left = cash_left_by_categ.mean()
+
+        if self.cash < 100:         #If their level of cash is so low that they cannot underwrite anything they also leave the market.
+            self.enter_bankruptcy(time)
+        else:
+            if self.is_insurer:
+
+                if len(self.underwritten_contracts) < 4 or avg_cash_left / self.cash > 0.6:   #Insurers leave the market if they only have 4 contracts underwritten or an excess capital of 60 percent for more than 24 periods.
+                    self.market_permanency_counter += 1
+                else:
+                    self.market_permanency_counter = 0                                    #All these limits maybe should be parameters in isleconfig.py
+
+                if self.market_permanency_counter >= 24:
+                    self.enter_bankruptcy(time)
+
+            if self.is_reinsurer:
+
+                if len(self.underwritten_contracts) < 2 or avg_cash_left / self.cash > 0.8:   #Reinsurers leave the market if they only have 2 contracts underwritten or an excess capital of 80 percent for more than 100 periods.
+
+                    self.market_permanency_counter += 1                                       #Insurers and reinsurers potentially have different reasons to leave the market. That's why the code is duplicated here.
+                else:
+                    self.market_permanency_counter = 0
+
+                if self.market_permanency_counter >= 200:
+                    self.enter_bankruptcy(time)
```

Comment:
This should not be count as bankruptcy. But I leave it like that for the moment.

Reply:
Maybe, including a TODO would help us remember this later.

Reply:
Done.